website/docs/language/expressions/references.mdx

```diff
@@ -118,14 +118,16 @@ to mark the reference as for a data resource.
 
 ### Filesystem and Workspace Info
 
-* `path.module` is the filesystem path of the module where the expression
-  is placed.
-* `path.root` is the filesystem path of the root module of the configuration.
-* `path.cwd` is the filesystem path of the current working directory. In
+The following values are available:
+
+- `path.module` is the filesystem path of the module where the expression
+  is placed. You can use `path.module` in write operations, but we do not recommend this when using local module sources. Multiple invocations of local modules use the same source directory, overwriting the data in `path.module` during each call. This can lead to race conditions and unexpected results.
```

Comment:
I would prefer for it to say:

```
It is not recommended to use `path.module` in write operations because you 
can receive different, unpredictable behavior depending on if you are using 
remote or local module sources. Multiple invocations of local modules use the 
same source directory, overwriting the data in `path.module` during each call. 
This can lead to race conditions and unexpected results.
```
Just to be stronger against using it for write operations. The reason for this is that many people use remote module source in production/automation, but for testing remote modules, will frequently switch the module source path to use local source instead so that they don't have to push up changes to test them. 

If I was reading the current wording, I would say "Oh, we don't use local module source so HashiCorp says I can do this" and move on without thinking about the fact that people often switch to local module source when iterating on the remote module locally. That's why I think the wording should be stronger and more along the lines of "Don't do this and here is why".

Reply:
You know, I see your point - thank you for commenting! I want to bring engineering on this because I believe they know a bit more about how this is used generally, and I want to make sure my opinion isn't the only opinion driving this.  "HashiCorp doesn't recommend X" are important, and I want to get another set of eyes on this.

Reply:
Sounds good. Also, hopefully this will all change when https://github.com/hashicorp/terraform/issues/29503 is addressed and it sounds like whenever the Terraform team addresses that issue, it will involve a breaking change to this behavior so it would be best to advise people to avoid something that is likely to break in the future.